contributed_definitions/nyaml/NXapm_compositionspace_config.yaml

```diff
@@ -34,61 +21,97 @@ NXapm_compositionspace_config(NXobject):
       analysis_identifier(NX_UINT):
         exists: recommended
       reconstruction(NXserialized):
+        doc: |
+          Specification of the tomographic reconstruction used for this analysis.
+          Typically, reconstructions in the field of atom probe tomography are communicated
+          via files which store at least reconstructed ion positions and mass-to-charge-state-ratio
+          values. Container files like HDF5 though can store multiple reconstructions.
+          Therefore, the position and mass_to_charge concepts point to specific instances
+          to use for this analysis.
         type(NX_CHAR):
         path(NX_CHAR):
         checksum(NX_CHAR):
         algorithm(NX_CHAR):
         position(NX_CHAR):
-        mass_to_charge(NX_CHAR):
+          doc: |
+            Name of the node which resolves the reconstructed ion position values to use for this analysis.
+        # mass_to_charge(NX_CHAR):
       ranging(NXserialized):
+        doc: |
+          Specification of the ranging definitions used for this analysis.
+          
+          Indices start from 1. The value 0 is reserved for the null model of unranged positions
+          whose iontype is unknown_type. The value 0 is alsoreserved for voxel outside the dataset.
         type(NX_CHAR):
         path(NX_CHAR):
         checksum(NX_CHAR):
         algorithm(NX_CHAR):
         ranging_definitions(NX_CHAR):
-      edge_length(NX_FLOAT):
+          doc: |
+            Name of the (parent) node directly below which the ranging definitions for (molecular) ions are stored.
+      voxelization(NXprocess):
         doc: |
-          Edge length of the cubic voxels that is used for discretizing the point cloud in the voxelization step.
-        unit: NX_LENGTH
-      bics_clusters(NX_UINT):
+          Step during which the point cloud is discretized to compute element-specific composition fields.
+          Iontypes are atomically decomposed to correctly account for the multiplicity of each element that
+          was ranged for each ion.
+        edge_length(NX_NUMBER):
+          doc: |
+            Edge length of cubic voxels for 3D grid that is used for discretizing the point cloud.
+          unit: NX_LENGTH
+      autophase(NXprocess):
         doc: |
-          The maximum number of components configured for the segmentation of the voxel
-          set into a mixture of voxels with that many different chemical compositions.
-        unit: NX_UNITLESS
-      ml_models(NXobject):
+          Optional step during which the subsequent segmentation step is prepared to improve the segmentation.
+        use(NX_BOOLEAN):
```

Comment:
maybe call it `used`. Alternative, make it the NXprocess optional/recommended and if it is there, it is assumed it was used.